lib/codecs/src/encoding/format/protobuf.rs

```diff
@@ -31,7 +31,7 @@ impl ProtobufSerializerConfig {
 
     /// The data type of events that are accepted by `ProtobufSerializer`.
     pub fn input_type(&self) -> DataType {
-        DataType::Log.and(DataType::Trace)
+        DataType::all()
```

Comment:
Can this codec encode metrics? That would surprise me a little. I think this codec may only be applicable to logs.

Reply:
Consider that the `protobuf` decoder only emits logs: https://github.com/vectordotdev/vector/blob/92d4102df59f3239b9aa062caa28e19ead7f3c57/lib/codecs/src/decoding/format/protobuf.rs#L41-L43